zerver/views/messages.py

```diff
@@ -511,7 +511,7 @@ def convert_term(elem: Union[Dict[str, Any], List[str]]) -> Dict[str, Any]:
         # We have to support a legacy tuple format.
         if isinstance(elem, list):
             if (len(elem) != 2 or
-                any(not isinstance(x, str) and not isinstance(x, str)
+                any(not isinstance(x, str) and not isinstance(x, str)  # type: ignore # ?
```

Comment:
This is very interesting. Why this type of check to see if the element is a string pair?

Reply:
This was mis-converted from `not isinstance(x, str) and not isinstance(x, Text)` in #9204. We can delete the redundant copy of the test.

Reply:
Ah I see.  I thought this was probably a redundancy, but wanted to give the benefit of the doubt. :)